web/shapeworks/src/components/TabForm.vue

```diff
@@ -0,0 +1,115 @@
+<script>
+import { defineComponent, ref } from '@vue/composition-api'
+import Vue from 'vue'
+import Vuetify from 'vuetify'
+import 'vuetify/dist/vuetify.min.css'
+import VJsf from '@koumoul/vjsf'
+import '@koumoul/vjsf/dist/main.css'
+import Ajv from 'ajv';
+import defaults from 'json-schema-defaults';
+import { submitForm } from '../store';
+
+Vue.use(Vuetify)
+
+
+export default defineComponent({
+    props: {
+        form: {
+            type: String,
+            required: true,
+        }
+    },
+    components: {
+      VJsf,
+    },
+    setup(props) {
+        const ajv = new Ajv();
+        const formDefaults = ref();
+        const formData = ref({});
+        const formSchema = ref();
+        const formValid = ref(false);
+        const refreshing = ref(false);
+
+        async function fetchFormSchema() {
+            formSchema.value = await (await fetch( `forms/${props.form}.json`)).json()
+            formDefaults.value = defaults(formSchema.value)
+        }
+        fetchFormSchema()
+
+        function evaluateExpression (expression) {
+            if(expression){
+                // The shemas are trusted source code
+                expression = eval(expression)
+                return expression(formData.value)
+            }
+            return true;
+        }
+
+        const schemaOptions = {
+            rootDisplay: "expansion-panels",
+            autoFocus: true,
+            ajv,
+            sliderProps: {
+                thumbLabel: true
+            }
+        }
+
+        function resetForm () {
+            refreshing.value = true;
+            formData.value = formDefaults.value;
+            setTimeout(() => { refreshing.value = false; }, 1)
+        }
+
+        return {
+            props,
+            formData,
+            schemaOptions,
+            formSchema,
+            formValid,
+            refreshing,
+            resetForm,
+            submitForm,
+            evaluateExpression,
+        }
+    },
+})
+</script>
+
+
+<template>
+<v-form v-model="formValid" class="pa-3" >
+    <div style="display: flex; width: 100%; justify-content: space-between;">
+        <v-btn @click="resetForm">Restore defaults</v-btn>
+        <v-btn color="primary" @click="() => { submitForm(props.form, formData) }">{{ props.form }}</v-btn>
+    </div>
+    <br />
+    <v-jsf
```

Comment:
For now I think this is perfectly fine. If we ever need more general nested recursion, I think it would be possible to create a `ConditionalVJSF` component that would looks something like this:
```
<template>
  <v-jsf ...>
    <template slot="custom-conditional" slot-scope="context">
      <ConditionalVJSF
        v-if="evaluateExpression(...)"
        ...
      />
    </template>
  </v-jsf>
</template>
```

It certainly feels like something that should work out of the box, but vjsf is still maturing.